### PR TITLE
docs(cicd): Add CI/CD integration guide

### LINKED ISSUE #2321: Write a "Automate your CI/CD with the Modeler API" guide
To promote the integration of Web Modeler into CI/CD pipelines using the API, we should have a guide that describes the options here.

### PR DESCRIPTION
**Pull request by christian-konrad**
## Description

Closes #2321 

## When should this change go live?

- [x] This functionality is already available but undocumented.

## PR Checklist

- [ ] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [x] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [ ] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [ ] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

**Comment by akeller** [comment]
Open question re: terms with the highest traction in this space - https://camunda.slack.com/archives/C0589TC88UU/p1692432381103939

**Comment by christian-konrad** [comment]
Hi @akeller and @christinaausley, this guide is ready for review. Happy about any feedback—it is quite a long read.

**Comment by christian-konrad** [comment]
@CatalinaMoisuc @wollefitz also happy to read your feedback!

**Comment by christinaausley** [comment]
@christian-konrad I'll be sure to have a look at this before the end of the week 👍

File: docs/guides/devops-lifecycle/img/visual-diff.png
Comment (christinaausley) [inline docs/guides/devops-lifecycle/img/visual-diff.png:1]: I *think* having your initials here in the screenshot is okay, but it is still private info. Thoughts?

File: docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd.md [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+
+Integrating Web Modeler into your CI/CD pipelines can significantly enhance process application development and deployment workflows. By automating process application deployment, changes can be promptly and accurately reflected in the production environment. This agility empowers teams to swiftly respond to evolving business needs, fostering a flexible and adaptable process orchestration approach.
+
+## Prerequisites
+
+Each pipeline is unique. The Web Modeler API offers flexibility to tailor integrations according to your pipelines. To get started, there are a few prerequisites based on your setup:
+
+- A version control system (VCS) such as GitHub or GitLab.
+- An existing pipeline or a plan to set one up using tools like [CircleCI](https://circleci.com/) or [Jenkins](https://www.jenkins.io/), cloud platforms such as [Azure DevOps Pipelines](https://azure.microsoft.com/de-de/products/devops), or built-in VCS platforms like [GitHub Actions](https://github.com/features/actions) or [GitLab's DevSecOps Lifecycle](https://about.gitlab.com/stages-devops-lifecycle/).
+- Make yourself familiar with the [Web Modeler API](/apis-tool/web-modeler-api) through the OpenAPI documentation.
```
Comment (christinaausley) [inline docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd.md:29]: Perhaps link to OpenAPI documentation?

File: docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd.md [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+
+If you don't need to spawn all apps such as Operate or Tasklist, you can use the lightweight [embedded Zeebe engine](https://github.com/camunda-community-hub/eze), which is a community-maintained project, to set up a cost-effective solution with an in-memory database. Together with the [Zeebe Hazelcast exporter](https://github.com/camunda-community-hub/zeebe-hazelcast-exporter) (community-maintained as well), you can consume data generated from your process for reporting or testing.
+
+In the build stage, deploy your process or project to a cluster or embedded engine. Post-pipeline completion, such as deployment to staging or production, preview environments can be discarded.
+
+:::tip
+For GitLab users, consider using [GitLab Review Apps](https://docs.gitlab.com/ee/ci/review_apps/) to provide preview environments.
+:::
+
+Deploy resources using the [`zbctl` CLI](/apis-tools/cli-client) in this pipeline step, compatible with both SaaS and Self-Managed clusters. Alternately, utilize the Java or Go client library or any community-built alternatives.
```
Comment (christinaausley) [inline docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd.md:196]: Do we want to link to Java/Go client libraries here, and community clients as well?

File: docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd.md [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+
+The following process diagram demonstrates an example flow of how to run a preview using milestones and a diff link in GitHub:
+
+<iframe src="https://modeler.cloud.ultrawombat.com/embed/35868bd2-a690-48de-a069-aa8ae6b3a846" style={{width: "100%", height: "500px", border: "1px solid #ccc"}} allowfullscreen></iframe>
+
+#### Review a running process application
+
+If deployed in a review environment, processes/applications can be shared with peers for interactive review. For comprehensive review, full clusters inclusive of Operate and Tasklist can be used for process execution. This closely simulates the final experience. To integrate the preview environment with custom applications, leverage the Operate and Tasklist APIs and deploy them within the review environment.
+
+In case you use an embedded Zeebe engine, or want to provide a lightweight, focused review experience, you can use [Zeebe Simple Monitor](https://github.com/camunda-community-hub/zeebe-simple-monitor), which is a community-maintained Web App similar to the Play mode in Web Modeler. Deploying Zeebe SimpleMonitor is easy, and allows for thorough process testing and review.
```
Comment (christinaausley) [inline docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd.md:268]: Opportunity to link to Play here.

File: docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd.md [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+
+Yes, you can use the `bpmnlint` and `dmnlint` libraries to automatically verify your process diagrams against predefined rules. These libraries provide reporting capabilities to identify and fix issues during the build stage.
+
+### How can I perform unit and integration tests on my processes?
+
+You can use the `zeebe-process-test` library for Java-based unit tests or community-built clients for other programming languages. These libraries allow you to execute your BPMN and DMN diagrams with assertions in your development or preview environments.
+
+### How do I provide environment variables to Connectors in preview environments?
+
+You can manage environment variables for Connectors using secrets. This can be set up in both Camunda Platform 8 SaaS and Self-Managed. Refer to the Connectors configuration documentation for details.
```
Comment (christinaausley) [inline docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd.md:323]: Would definitely link to Connectors config docs here.

**Review by christinaausley — DISMISSED** [review]
Technical review in latest commits, and some comments for opportunity to provide links.

File: docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd.md [hunk 5]
```diff
@@ -0,0 +1,332 @@
...
+
+Integrating Web Modeler into your CI/CD pipelines can significantly enhance process application development and deployment workflows. By automating process application deployment, changes can be promptly and accurately reflected in the production environment. This agility empowers teams to swiftly respond to evolving business needs, fostering a flexible and adaptable process orchestration approach.
+
+## Prerequisites
+
+Each pipeline is unique. The Web Modeler API offers flexibility to tailor integrations according to your pipelines. To get started, there are a few prerequisites based on your setup:
+
+- A version control system (VCS) such as GitHub or GitLab.
+- An existing pipeline or a plan to set one up using tools like [CircleCI](https://circleci.com/) or [Jenkins](https://www.jenkins.io/), cloud platforms such as [Azure DevOps Pipelines](https://azure.microsoft.com/de-de/products/devops), or built-in VCS platforms like [GitHub Actions](https://github.com/features/actions) or [GitLab's DevSecOps Lifecycle](https://about.gitlab.com/stages-devops-lifecycle/).
```
Comment (wollefitz) [inline docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd.md:28]:
```suggestion
- An existing pipeline or a plan to set one up using tools like [CircleCI](https://circleci.com/) or [Jenkins](https://www.jenkins.io/), cloud platforms such as [Azure DevOps Pipelines](https://azure.microsoft.com/de-de/products/devops), or built-in solutions of VCS platforms like [GitHub Actions](https://github.com/features/actions) or [GitLab's DevSecOps Lifecycle](https://about.gitlab.com/stages-devops-lifecycle/).
```

File: docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd.md [hunk 6]
```diff
@@ -0,0 +1,332 @@
...
+- Make yourself familiar with the [Web Modeler API](/apis-tools/web-modeler-api/index.md) through the [OpenAPI documentation](https://modeler.cloud.camunda.io/swagger-ui/index.html).
+- Understand how [clusters](http://localhost:3000/docs/next/components/concepts/clusters/) work in Camunda Platform 8.
```
Comment (wollefitz) [inline docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd.md:30]:
```suggestion
- Understand how [clusters](/docs/next/components/concepts/clusters/) work in Camunda Platform 8.
```

File: docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd.md [hunk 7]
```diff
@@ -0,0 +1,332 @@
...
+
+<Tabs groupId="disableDeployments" defaultValue="sm" values={[{label: 'Self-Managed', value: 'sm', }, {label: 'SaaS', value: 'saas', },]} >
+<TabItem value="sm">
+
+You can completely disable manual deployments by configuring environment variables `ZEEBE_BPMN_DEPLOYMENT_ENABLED` and `ZEEBE_DMN_DEPLOYMENT_ENABLED` as documented [here](/self-managed/modeler/web-modeler/configuration/configuration.md/#general).
+
+</TabItem>
+<TabItem value="saas">
+
+Currently, manual deployments are unrestricted on SaaS. Efforts are underway to introduce permissions for restricting deployments.
```
Comment (wollefitz) [inline docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd.md:62]: We could mention here that only the `Developer` role allows deployments from Web Modeler. Assigning any other role effectively removes deployment privileges, see 

https://github.com/camunda/camunda-platform-docs/blob/01845885995f2a91908aeeb88dccc7dd0a9483ba/docs/components/console/manage-organization/manage-users.md?plain=1#L25-L29

File: docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd.md [hunk 8]
```diff
@@ -0,0 +1,332 @@
...
+
+#### Can I implement blue-green deployments with Camunda Platform 8 SaaS?
+
+While blue-green deployments are more straightforward with Self-Managed setups, you can implement similar deployment strategies with Camunda Platform 8 SaaS. Keep in mind the limitations and differences between clusters when planning your deployment approach.
+
+#### How can I prevent manual deployments from Web Modeler?
+
+To enforce CI/CD pipelines and restrict manual deployments, you can disable manual deployments. For Self-Managed deployments, set environment variables `ZEEBE_BPMN_DEPLOYMENT_ENABLED` and `ZEEBE_DMN_DEPLOYMENT_ENABLED`. Manual deployments on SaaS can't be restricted currently, but permissions to restrict deployments are being worked on.
```
Comment (wollefitz) [inline docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd.md:299]: See above

File: docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd.md [hunk 9]
```diff
@@ -0,0 +1,332 @@
...
+
+Continuous integration and deployment are pivotal for rapid and reliable software development, testing, and delivery. These practices automate the building, testing, and deployment processes, leading to shorter development cycles, enhanced collaboration, and higher-quality releases.
+
+Integrating Web Modeler into your CI/CD pipelines can significantly enhance process application development and deployment workflows. By automating process application deployment, changes can be promptly and accurately reflected in the production environment. This agility empowers teams to swiftly respond to evolving business needs, fostering a flexible and adaptable process orchestration approach.
+
+## Prerequisites
+
+Each pipeline is unique. The Web Modeler API offers flexibility to tailor integrations according to your pipelines. To get started, there are a few prerequisites based on your setup:
+
+- A version control system (VCS) such as GitHub or GitLab.
```
[the remark this answers is not in the document]
Comment (christian-konrad) [reply docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd.md:27]: Yeah, you are right, I will update this!

File: docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd.md (hunk 6 above)
Comment (christian-konrad) [reply docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd.md:30]: Good catch, thanks!

File: docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd.md (hunk 7 above)
Comment (christian-konrad) [reply docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd.md:62]: You're right! Totally missed that. Thanks!

**Comment by christian-konrad** [comment]
@christinaausley I included all the review feedback, could you please give it a look and approve it so I can merge?

**Review by christinaausley — APPROVED** [review]

**Comment by akeller** [comment]
@christian-konrad, I did not have a chance to review this this week. A few notes before I sign off for the weekend -

* As it stands in this PR, you've only made changes to 8.3 or `/next/`, but your description indicates this is an undocumented but available feature. Please backport to 8.2 by adding your changes to `/versioned_docs/version-8.2/` and `versioned_sidebars/version-8.2-sidebars.json`.
* Currently the Guides section is split between **Getting started** and **Next steps** as of the initial [Docs Visions](https://docs.google.com/presentation/d/122T4xcwOejuFbhQu0KpoAzL-rwCkaX6bUirym3IL_DE/edit?pli=1#slide=id.g14510d13a93_0_20). You can keep your sidebar structure (although I would recommend against folders with only one topic, unless you plan to have more in the near future), but you will need to move it into **Next steps**. 
  * If you disagree with this structure, approach me so we can initiate a thoughtful restructure of this content, something that is already on the radar and part of the review work @christinaausley is leading - https://github.com/camunda/developer-experience/issues/96#issuecomment-1684279742

**Comment by christian-konrad** [comment]
Hi @akeller , thanks for the comment!
So far, the guide describes features only available in 8.3 alphas, so I don't want to backport it to 8.2.

On the structure, I am fine to move it to 'next steps' and find another way to increase awareness of this guide

File: docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd.md [hunk 10]
```diff
@@ -0,0 +1,332 @@
...
+{
+  "filter": {
+    "projectId": "<PROJECT TO SYNC>",
+    "updated": "<LAST SYNC DATE>",
+  ],
+  "page": 0,
+  "size": 50
+}
+```
+-->
```
Comment (wollefitz) [inline docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd.md:95]: @christian-konrad As https://github.com/camunda/web-modeler/issues/5754 is now merged filtering by date ranges is now possible. Documentation is still pending but it basically works the same way as for the other APIs by appending a date range suffix like `||/m` to the date string, see also https://docs.camunda.io/docs/next/apis-tools/operate-api/overview/#filter-dates.

**Pull request merged**